scipy/_lib/_array_api.py

```diff
@@ -56,6 +56,16 @@ def compliance_scipy(arrays: list[ArrayLike]) -> list[Array]:
     """
     for i in range(len(arrays)):
         array = arrays[i]
+
+        # this comes from `_util._asarray_validated`
+        if not sparse_ok:
+            import scipy.sparse
+            if scipy.sparse.issparse(array):
```

Comment:
perhaps we'd rather do this import at the top of the module - I just copied the implementation over